src/composables/fungibleTokens.ts

```diff
@@ -0,0 +1,307 @@
+import { ref, watch } from 'vue';
+import camelCaseKeysDeep from 'camelcase-keys-deep';
+import BigNumber from 'bignumber.js';
+import { Encoded, Encoding } from '@aeternity/aepp-sdk';
+import { fetchAllPages, handleUnknownError, toShiftedBigNumber } from '@/utils';
+import type {
+  BigNumberPublic, IDefaultComposableOptions,
+  IToken,
+  ITokenList,
+  ITransaction,
+} from '@/types';
+import { PROTOCOL_AETERNITY, TX_DIRECTION } from '@/constants';
+import { aettosToAe, calculateSupplyAmount, categorizeContractCallTxObject } from '@/protocols/aeternity/helpers';
+import FungibleTokenFullInterfaceACI from '@/lib/contracts/FungibleTokenFullInterfaceACI.json';
+import AedexV2PairACI from '@/lib/contracts/AedexV2PairACI.json';
+import ZeitTokenACI from '@/lib/contracts/FungibleTokenFullACI.json';
+import { AE_SYMBOL } from '@/protocols/aeternity/config';
+import { useAccounts } from './accounts';
+import { useAeSdk } from './aeSdk';
+import { useMiddleware } from './middleware';
+import { useTippingContracts } from './tippingContracts';
+import { createNetworkWatcher } from './networks';
+import { createPollingBasedOnMountedComponents } from './composablesHelpers';
+
+const availableTokens = ref<ITokenList>({});
+const tokenBalances = ref<Record<string, IToken[]>>({});
+
+const { onNetworkChange } = createNetworkWatcher();
+const availableTokensWatcher = createPollingBasedOnMountedComponents(60000);
+const tokenBalancesWatcher = createPollingBasedOnMountedComponents(10000);
```

Comment:
Those functions are not watchers for either availableTokens nor tokenBalances - they initialises polling